ui/src/app/nav.tsx

```diff
@@ -0,0 +1,134 @@
+"use client"
+
+import { SetStateAction, Dispatch, useState, createContext, useEffect } from "react"
+import Image from "next/image";
+
+type AsideContextType = {
+    isOpen: boolean,
+    setIsOpen: Dispatch<SetStateAction<boolean>>
+}
+export const AsideContext = createContext<AsideContextType>({ isOpen: false, setIsOpen: () => { } });
+
+export function Aside({ isOpen, setIsOpen }: { isOpen: boolean, setIsOpen: Dispatch<SetStateAction<boolean>> }) {
+    return (
+        <aside className={"l-aside" + (isOpen ? "" : " is-collapsed")} id="aside-panel" aria-label="aside-panel">
+            <div className="p-panel">
+                <div className="p-panel__header">
+                    <h4 className="p-panel__title">Add New CSR</h4>
+                    <div className="p-panel__controls">
+                        <button onClick={() => setIsOpen(false)} className="p-button--base u-no-margin--bottom has-icon"><i className="p-icon--close"></i></button>
+                    </div>
+                </div>
+                <div className="p-panel__content">
+                    <form className="p-form p-form--stacked">
+                        <div className="p-form__group row">
+                            <label htmlFor="textarea">
+                                Enter or upload CSR in PEM format below
+                            </label>
+                        <div className="p-form__group row">
+                            <button className="p-button u-float-left" name="upload">Upload</button>
+                        </div>
+                            <textarea id="textarea" name="textarea" rows={10} placeholder="-----BEGIN CERTIFICATE REQUEST-----"></textarea>
+                        </div>
+                        <div className="p-form__group row">
+                            <button className="p-button--positive u-float-right" name="submit">Submit</button>
+                        </div>
+                    </form>
+                </div>
+            </div >
+        </aside >
+    )
+}
+
```

Comment:
Once we have a bunch of modals we may end up preferring to have a modal for this too, but most examples of Aside's in the vanilla docs use it for this exact purpose, to submit a form.

Reply:
I'm fine with it being an aside, what I'm not sure is the layout inside of it.

Reply:
I'd consider this part of the ticket to make this thing work, since there's a bunch of smart things we could do here. Off the top of my head:
* [validation](https://vanillaframework.io/docs/base/forms#validation)
* live CSR details text
* picking multiple files

I'm also not sure about the file picker. It was the first thing I'd done, but it looks so out of place and if we just paste the content of the file into the box it's redundant to show the filename. We can actually [disable the vanilla upload button](https://vanillaframework.io/docs/patterns/buttons#default) immediately after the upload completes. We could also style the input field with vanilla. CSS overrides standard styles that browsers have for \</input> tags for example if we add styling that modifies the same css property. I did some weird stuff with it now (subject to change later). check it out.

but i can also incorporate this look if we want it.

Reply:
I think the Vanilla file picker is just fine honestly, this is exactly what it's made for. The current box-in-a-box/button-in-a-button is strange though.

![image](https://github.com/canonical/gocert/assets/18486508/2a797150-5e40-460a-a9c2-63c999baee5c)

Reply:
ok, changed it back. I got a bit carried away so now you can only upload a .pem file by default and also that value gets pasted into this text area immediately. You can then edit it or upload another file which will overwrite the text area.

If I don't merge this thing I'm gonna finish the frontend at this point :D